### ISSUE TITLE
Error NVIDIA-SMI couldn't find libnvidia-ml.so library in your system (nvidia/cuda-ppc64le:8.0-cudnn6-devel-ubuntu16.04)

### ISSUE THREAD

**Issue by sandipmgiri** [NVIDIA/nvidia-docker#546]
When I use the command from the host: nvidia-smi

I get this result:
```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 384.81                 Driver Version: 384.81                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla P100-SXM2...  Off  | 00000002:01:00.0 Off |                    0 |
| N/A   28C    P0    29W / 300W |     10MiB / 16276MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla P100-SXM2...  Off  | 00000006:01:00.0 Off |                    0 |
| N/A   27C    P0    30W / 300W |     10MiB / 16276MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

```

But when I run `sudo nvidia-docker run -it nvidia/cuda-ppc64le:8.0-cudnn6-devel-ubuntu16.04 nvidia-smi`

Result:
```
NVIDIA-SMI couldn't find libnvidia-ml.so library in your system. Please make sure that the NVIDIA Display Driver is properly installed and present in your system.
Please also try adding directory that contains libnvidia-ml.so to your system PATH.
```

Looks like NVIDIA Display Driver is not already installed/available in nvidia/cuda-ppc64le:8.0-cudnn6-devel-ubuntu16.04 image.

This issue got resolved after installing nvidia-384-dev and now `nvidia-smi` command working fine (GPUs are accessible) : 
```
1) sudo nvidia-docker run -it nvidia/cuda-ppc64le:8.0-cudnn6-devel-ubuntu16.04 /bin/bash
2) apt-get install nvidia-384-dev
3) nvidia-smi
```

Is it mandatory to install nvidia-384-dev? There is any other solution ? or I am missing something here ? 

Basically we are trying use the base image , in which CUDA, cuDNN and GPUs are accessible. (for ubuntu16.04_ppc64le)

**Comment by 3XX0** [comment]
No you shouldn't install the driver inside the container.
Assuming you have `nvidia-docker` 1.0 installed, what the output of `sudo journalctl -u nvidia-docker`

**Comment by sandipmgiri** [comment]
I have installed version `nvidia-docker-1.0.1-1.ppc64le` .
Please find attached file for output of `sudo journalctl -u nvidia-docker`
[output.txt](https://github.com/NVIDIA/nvidia-docker/files/1517536/output.txt)

**Comment by 3XX0** [comment]
Can you check the content of `/usr/local/nvidia` inside the container to see if the library is there.

**Comment by sandipmgiri** [comment]
```
$ ls /usr/local/nvidia
bin
```

Only `/bin` directory is there , no library available inside the container

**Comment by 3XX0** [comment]
That's weird, your volume might be altered. try removing the volume `docker volume rm nvidia_driver_384.81` and try running the container again.

**Comment by sandipmgiri** [comment]
Initially for `nvidia-docker` I was getting following error -
```
$ sudo nvidia-docker run -it  nvidia/cuda-ppc64le:8.0-cudnn6-devel-ubuntu16.04 nvidia-smi

docker: Error response from daemon: create nvidia_driver_384.81: VolumeDriver.Create: internal error, check logs for details.
See 'docker run --help'.
```
```
$ ls -lR /var/lib/nvidia-docker/volumes

/var/lib/nvidia-docker/volumes:
total 0
```
 To resolve this I ran following commands manually - 
```
sudo mkdir -p /var/lib/nvidia-docker/volumes/nvidia_driver/384.81/bin
sudo cp /usr/bin/nvidia-* /var/lib/nvidia-docker/volumes/nvidia_driver/384.81/bin/
```

Would that affect?

**Comment by 3XX0** [comment]
Unfortunately this is a [known limitation](https://github.com/NVIDIA/nvidia-docker/wiki/nvidia-docker-plugin#known-limitations) of 1.0

**Comment by sandipmgiri** [comment]
Hi @3XX0 , Can you please give some details on this to proceed further.Thanks!

**Comment by 3XX0** [comment]
Until we have 2.0 for PPC64, you can use one of the workaround described in the link above (for [example](https://github.com/NVIDIA/nvidia-docker/issues/216#issuecomment-252800776))

**Comment by sandipmgiri** [comment]
Thank you for your response, @3XX0.

I have used the following workaround  :
		
		$ systemctl edit nvidia-docker
		
		# add following:
		
		[Service]
		ExecStart=
		ExecStart=/usr/bin/nvidia-docker-plugin -s $SOCK_DIR -d /usr/local/nvidia-driver		  
	
		$ sudo chown nvidia-docker  /usr/local/nvidia-driver
		$ sudo systemctl restart docker nvidia-docker


Now the issue got resolved .
Inside the container : `Library available` and `GPUs are accessible` :

```
$ ls /usr/local/nvidia/lib64/
libEGL.so.1               libGLESv1_CM_nvidia.so.384.81  libGLdispatch.so.0              libnvidia-fatbinaryloader.so.384.81  libnvidia-ptxjitcompiler.so.1
libEGL_nvidia.so.0        libGLESv2.so.2                 libOpenGL.so.0                  libnvidia-glcore.so.384.81           libnvidia-ptxjitcompiler.so.384.81
libEGL_nvidia.so.384.81   libGLESv2_nvidia.so.2          libcuda.so                      libnvidia-glsi.so.384.81             libnvidia-tls.so.384.81
libGL.so.1                libGLESv2_nvidia.so.384.81     libcuda.so.1                    libnvidia-ml.so.1
libGL.so.384.81           libGLX_indirect.so.0           libcuda.so.384.81               libnvidia-ml.so_bkup
libGLESv1_CM.so.1         libGLX_nvidia.so.0             libnvidia-egl-wayland.so.1.0.1  libnvidia-opencl.so.1
libGLESv1_CM_nvidia.so.1  libGLX_nvidia.so.384.81        libnvidia-eglcore.so.384.81     libnvidia-opencl.so.384.81

```
```
$  sudo nvidia-docker run -it   nvidia/cuda-ppc64le:8.0-cudnn6-devel-ubuntu16.04 nvidia-smi
	
Wed Dec  6 06:45:49 2017
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 384.81                 Driver Version: 384.81                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla P100-SXM2...  Off  | 00000002:01:00.0 Off |                    0 |
| N/A   27C    P0    29W / 300W |     10MiB / 16276MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla P100-SXM2...  Off  | 00000006:01:00.0 Off |                    0 |
| N/A   27C    P0    29W / 300W |     10MiB / 16276MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```

Thank you very much for your help and suggestions!!
